aten/src/ATen/native/DilatedMaxPool2d.cpp

```diff
@@ -58,12 +50,83 @@ void max_pool2d_with_indices_out_cpu_template(
   const int64_t outputHeight = pooling_output_shape<int64_t>(inputHeight, kH, padH, dH, dilationH, ceil_mode);
   const int64_t outputWidth = pooling_output_shape<int64_t>(inputWidth, kW, padW, dW, dilationW, ceil_mode);
 
+  const auto memory_format = input.suggest_memory_format();
   pool2d_shape_check(
     input,
     kH, kW, dH, dW, padH, padW, dilationH, dilationW,
     nInputPlane,
     inputHeight, inputWidth,
-    outputHeight, outputWidth, input.suggest_memory_format());
+    outputHeight, outputWidth, memory_format);
+
+  /* resize output and indices */
+  if (input.ndimension() == 3) {
+    if (memory_format == at::MemoryFormat::Contiguous) {
+      set_output(0, {nInputPlane, outputHeight, outputWidth}, {outputHeight * outputWidth, outputWidth, 1}, input.options(), input.names());
+      /* indices will contain the locations for each output point */
+      set_output(1, {nInputPlane, outputHeight, outputWidth}, {outputHeight * outputWidth, outputWidth, 1}, input.options().dtype(kLong), input.names());
```

Comment:
Manually spelling out the strides here is a bad look. `set_output` is supposed to accept memory format via the TensorOptions argument; you can just leave strides as an empty array. Is there a reason this didn't work? Then you don't need to case on memory format anymore either.

Reply:
I will do as you say, thanks for explaining